tests/jballerina-unit-test/src/test/resources/test-src/types/xml/xml-literals.bal

```diff
@@ -395,4 +395,7 @@ function testQueryInXMLTemplateExpr() {
 
     string str3 = "<doc><doc><num>1</num><num>2</num><num>3</num></doc></doc>";
     test:assertEquals(x6.toString(), str3);
+
+    var x7 = xml `<doc>${(from string s in ["a", "b", "c"] select xml `${s}z`)}</doc>`;
+    test:assertEquals(x7.toString(), "<doc>azbzcz</doc>");
```

Comment:
Shall we add some tests for when the query expression is a nested expression in another kind of expression, like we discussed during the call?

Reply:
[1].
```bal
    xml x = xml `foo`;
    xml y = x + from string s in ["a", "b"] select xml `${s}`; // operator '+' not defined for 'xml' and 'xml:Text[]'
```

[2].
```bal
var y = xml `foo` + <xml>from string s in ["a", "b"] select xml `${s}`;
```

Well, I believe we discussed something like above.
[1] does not work, as we infer the rhs-expr type and it will be `xml:Text[]`.
[2] works even without this fix. (if we add a test it only tests the type cast type checking logic)

Reply:
Could even be error scenarios.

What I meant was something like

```ballerina
var x7 = xml `<doc>${xml `foo` + (from string s in ["a", "b", "c"] select xml `${s}z`)}</doc>`;
```

But yeah, this doesn't work atm either. Shall we create an issue for this?